OpenSim/Common/Exception.cpp

```diff
@@ -67,6 +69,23 @@ exception()
 //#endif
 }
 
+Exception::Exception(const std::string& file,
+                     size_t line,
+                     const std::string& func) {
+    addMessage(file + ":" + std::to_string(line) + _msgSeparator + 
```

Comment:
Also, I don't see why this is not flexible. Reversing the message is just a matter of changing implementation of `addMessage`.